### PR TITLE
fix broken gracefull shutdown

### PR DESCRIPTION
**Pull request by ekneg54**
testing this is hard. too hard for me and so I decided to not add a tests for this.

test the fix explorative with:

```bash
logprep  run ./quickstart/exampledata/config/http_pipeline.yml
```

then press `CRTL + C`

you should get this:

```
^C2024-04-19 15:41:18,980 Logprep Runner INFO    : Shutting down
2024-04-19 15:41:18,980 Logprep Runner INFO    : Initiated shutdown
2024-04-19 15:41:18,980 Logprep Pipeline 2 DEBUG   : Stopping pipeline (MainProcess)
2024-04-19 15:41:18,980 Logprep Pipeline 2 DEBUG   : Stopping pipeline (MainProcess)
2024-04-19 15:41:20,659 Logprep Pipeline 1 DEBUG   : Stopping pipeline (MainProcess)
2024-04-19 15:41:20,659 Logprep Pipeline 1 DEBUG   : Stopping pipeline (MainProcess)
2024-04-19 15:41:25,673 Logprep HTTPServer INFO    : Shutting down
2024-04-19 15:41:25,778 Logprep HTTPServer ERROR   : Cancel 0 running task(s), timeout graceful shutdown exceeded
2024-04-19 15:41:25,779 Logprep HTTPServer INFO    : Finished server process [229968]
2024-04-19 15:41:25,788 Prometheus Exporter INFO    : Cleaned up /tmp/logprep1
2024-04-19 15:41:25,791 Logprep Runner INFO    : Shutdown complete
```

### CODE REVIEW

**Comment by ekneg54** [comment]
the strange thing is that it depends on how logprep is startet.

by starting it with this the problem occurs:

```
logprep run ./quickstart/exampledata/config/pipeline.yml    
```

by starting it the old way the old gracefull shutdown behaviour is triggered:

```
$env:PYTHONPATH="."; python ./logprep/run_logprep.py run ./quickstart/exampledata/config/pipeline.yml
```

**Comment by ekneg54** [comment]
the problem was, that the startup script installed via pyproject.toml calls the `cli()` function directly. So the signal handlers were never registered and the graceful shutdown behaviour was lost.

File: logprep/connector/http/input.py [hunk 1]
```diff
@@ -364,12 +364,14 @@ def __init__(self, name: str, configuration: "HttpConnector.Config", logger: Log
...
             "loop": "asyncio",
-            "timeout_graceful_shutdown": 0,
+            "timeout_graceful_shutdown": 5,
         }
         self._config.uvicorn_config.update(internal_uvicorn_config)
         self.port = self._config.uvicorn_config["port"]
         self.host = self._config.uvicorn_config["host"]
         self.target = f"http://{self.host}:{self.port}"
+        if self.messages is None:  # this is only for testing. Is set by pipeline_manager
+            self.messages = mp.Queue(maxsize=self._config.message_backlog_size)
```
Comment (dtrai2) [inline logprep/connector/http/input.py:374]: If we only need this in testing, can't we also set the attribute in the corresponding test class then? Besides, I just executed the test without these lines and all the test run just fine.

File: logprep/run_logprep.py [hunk 2]
```diff
@@ -60,6 +58,9 @@ def cli() -> None:
     Logprep allows to collect, process and forward log messages from various data sources.
     Log messages are being read and written by so-called connectors.
     """
+    if "pytest" not in sys.modules:  # needed for not blocking tests
+        signal.signal(signal.SIGTERM, signal_handler)
+        signal.signal(signal.SIGINT, signal_handler)
```
Comment (dtrai2) [inline logprep/run_logprep.py:63]: How about we add a test that checks if these signals actual work as promised? Like running logprep with `process = subprocess.Popen()`, `process.send_signal()` and then checking if the process actually terminated as expected?

Comment (ekneg54) [reply logprep/run_logprep.py:63]: I tried this, but it did not work. 
I guess the reason is, that the logprep process name if started with subprocess is not "MainProcess". So the triggers do not work.

File: logprep/connector/http/input.py (hunk 1 above)
Comment (ekneg54) [reply logprep/connector/http/input.py:374]: in my setup the tests hang if run without these lines. So the ci pipline tests deadlocks also.

Comment (dtrai2) [reply logprep/connector/http/input.py:374]: ah okay strange, but okay. Thanks for clearing that up for me.

File: logprep/run_logprep.py (hunk 2 above)
Comment (dtrai2) [reply logprep/run_logprep.py:63]: too bad. but good to know that you tried it already, thanks! :+1:

**Review by dtrai2 — APPROVED** [review]

**Pull request merged**